pkg/kubelet/status/status_manager.go

```diff
@@ -868,9 +875,29 @@ func mergePodStatus(oldPodStatus, newPodStatus v1.PodStatus, couldHaveRunningCon
 			podConditions = append(podConditions, c)
 		}
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		// update pod disruption conditions if a new one is present, otherwise preserve the old one
+		cType := v1.AlphaNoCompatGuaranteeDisruptionTarget
```

Comment:
As discussed offline the result of this approach is that once the condition is added it will always continue to be persisted on the pod status. This is ok for now because we only set the condition for terminal pods, but this assumption may change in the future. Additionally, as we chatted offline there is concern of resulting in an additional status update (of just the condition) and later a second status update with the terminal phase. 

I would suggest we wrap this logic with check if the pod is attempting to transition to terminal phase instead of the "revert logic" below. 

This way we will only set the condition of the pod phase is terminal and only send a (single) combined status update with the pod being in terminal phase + the condition. 

Something like this:

```
if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
    // update pod disruption conditions if a new one is present, otherwise preserve the old one
    cType := v1.AlphaNoCompatGuaranteeDisruptionTarget

    _, oldCondition = podutil.GetPodConditionFromList(oldPodStatus.Conditions, cType)
    _, newCondition := podutil.GetPodConditionFromList(newPodStatus.Conditions, cType)

    if newCondition != nil && podutil.IsPodPhaseTerminal(newPodStatus.Phase) && !couldHaveRunningContainers {
        podConditions = append(podConditions, *newCondition)
    } else if oldCondition != nil {
        podConditions = append(podConditions, *oldCondition)
    }
}
```

Reply:
as below, the main idea was to avoid duplication of the logic around checking the transition and for `couldHaveRunningContainers`, also consistency with the current approach.

Reply:
@bobbypage applied your suggestion, after some modifications. In particular, extracted the `transitioningToTerminalPhase` bool for reuse.